--- a/src/network-mysqld-packet.c	2018-01-25 08:17:24.252671993 +0000
+++ b/src/network-mysqld-packet.c	2018-01-25 08:18:37.046671004 +0000
@@ -1144,7 +1144,7 @@
 	shake->challenge = g_string_sized_new(20);
 	shake->capabilities = 
 		CLIENT_PROTOCOL_41 |
-		CLIENT_SECURE_CONNECTION |
+		CLIENT_RESERVED2 |
 		0;
 
 
@@ -1218,7 +1218,7 @@
 	
 	err = err || network_mysqld_proto_skip(packet, 13);
 	
-	if (shake->capabilities & CLIENT_SECURE_CONNECTION) {
+	if (shake->capabilities & CLIENT_RESERVED2) {
 		err = err || network_mysqld_proto_get_string_len(packet, &scramble_2, 12);
 		err = err || network_mysqld_proto_skip(packet, 1);
 	}
@@ -1320,7 +1320,7 @@
 	auth->response = g_string_new("");
 	auth->username = g_string_new("");
 	auth->database = g_string_new("");
-	auth->capabilities = CLIENT_SECURE_CONNECTION | CLIENT_PROTOCOL_41;
+	auth->capabilities = CLIENT_RESERVED2 | CLIENT_PROTOCOL_41;
 
 	return auth;
 }
@@ -1375,7 +1375,7 @@
 		err = err || network_mysqld_proto_skip(packet, 23);
 	
 		err = err || network_mysqld_proto_get_gstring(packet, auth->username);
-		if (auth->capabilities & CLIENT_SECURE_CONNECTION) {
+		if (auth->capabilities & CLIENT_RESERVED2) {
 			err = err || network_mysqld_proto_get_lenenc_gstring(packet, auth->response);
 		} else {
 			err = err || network_mysqld_proto_get_gstring(packet, auth->response);
--- a/plugins/admin/admin-plugin.c	2018-01-25 08:19:23.309670375 +0000
+++ b/plugins/admin/admin-plugin.c	2018-01-25 08:19:32.180670255 +0000
@@ -200,7 +200,7 @@
 	challenge->server_version_str = g_strdup("5.0.99-agent-admin");
 	challenge->server_version     = 50099;
 	challenge->charset            = 0x08; /* latin1 */
-	challenge->capabilities       = CLIENT_PROTOCOL_41 | CLIENT_SECURE_CONNECTION | CLIENT_LONG_PASSWORD;
+	challenge->capabilities       = CLIENT_PROTOCOL_41 | CLIENT_RESERVED2 | CLIENT_LONG_PASSWORD;
 	challenge->server_status      = SERVER_STATUS_AUTOCOMMIT;
 	challenge->thread_id          = 1;
 
